library/core/src/iter/adapters/zip.rs

```diff
@@ -91,7 +91,7 @@ where
     #[doc(hidden)]
     unsafe fn __iterator_get_unchecked(&mut self, idx: usize) -> Self::Item
     where
-        Self: TrustedRandomAccess,
+        Self: TrustedRandomAccessNoCoerce,
```

Comment:
This means that get_unchecked can't be used on an adapter chain containing Zip when something in the chain is only `TrustedRandomAccess`. Isn't the problem the stepping implementations (such as `next()`) and not the get_unchecked?

Reply:
Nevermind, I got the inheritance wrong.